### PR TITLE
fix(agents): align RAI assessments with supplied materials

### PR DESCRIPTION
**Pull request by okada-kai**
# Pull Request

### Problems and fixes

* **Late input discovery:** Assessments could start before confirming available
  project materials and optional templates, causing rework or missing evidence.
  **Fix:** Phase 1 checks optional templates before project materials and
  permitted evidence sources.
* **Incomplete template output:** Sample rows or widgets could limit generated
  content to only part of the assessment. **Fix:** Supplied templates use a
  stable-ID `assessment-content.md` projection, and rows and widgets expand
  beyond existing placeholders.
* **Unprofessional report style:** Reports could include conversational,
  drafting, or tool narration that required cleanup. **Fix:** Generated
  documents use assessment-focused prose and exclude process narration.

## Related Issue(s)

None.

## Type of Change

Select all that apply:

**Code & Documentation:**

* [x] Bug fix (non-breaking change fixing an issue)
* [ ] New feature (non-breaking change adding functionality)
* [x] Breaking change (fix or feature causing existing functionality to change)
* [ ] Documentation update

**Infrastructure & Configuration:**

* [ ] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [ ] Security configuration
* [ ] DevContainer configuration
* [ ] Dependency update

**AI Artifacts:**

* [x] Reviewed contribution with `hve-builder` and addressed all actionable findings
* [x] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [x] Copilot agent (`.github/agents/*.agent.md`)
* [ ] Copilot skill (`.github/skills/*/SKILL.md`)
* [ ] Copilot hook (`.github/hooks/*/*.json`)
* [ ] Eval spec added/updated for changed AI artifacts (`evals/`)

> Note for AI Artifact Contributors:
>
> * Agents: Research, indexing/referencing other project (using standard VS Code GitHub Copilot/MCP tools), planning, and general implementation agents likely already exist. Review `.github/agents/` before creating new ones.
> * Skills: Must include both bash and PowerShell scripts. See [Skills](../docs/contributing/skills.md).
> * Model Versions: Contributions **MUST** target models listed in the model catalog (`scripts/linting/model-catalog.json`) whose provider appears in `providerAllowlist` and whose status is `ga` or `preview`. Run `npm run lint:models` to validate references.
> * See [Agents Not Accepted](../docs/contributing/custom-agents.md#agents-not-accepted) and [Model Version Requirements](../docs/contributing/ai-artifacts-common.md#model-version-requirements).

**Other:**

* [ ] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Sample Prompts (for AI Artifact Contributions)

**User Request:**

> Create an RAI assessment from project evidence and populate the supplied
> impact-assessment document and Mural templates.

**Execution Flow:**

1. Uses the existing entry mode and six-phase workflow.
2. Resolves the project slug and output requirements, then checks all optional
   document and Mural templates before project materials.
3. Persists each supplied template reference and creates the required
   `assessment-content.md` skeleton.
4. Populates the projection from project evidence without limiting repeatable
   rows or Mural widgets to existing placeholders.
5. Records authoritative findings in `rai-plan.md` and prepares only the
   outputs produced by the assessment.

**Output Artifacts:**

* `.copilot-tracking/rai-plans/{project-slug}/rai-plan.md` containing the
  canonical assessment findings, evidence, controls, decisions, and tradeoffs.
* `.copilot-tracking/rai-plans/{project-slug}/assessment-content.md`
  containing stable rendering IDs and the combined structure of every supplied
  template.
* Populated document and Mural outputs when requested.
* Reviewed and sanitized backlog handoff artifacts after confirmation.

**Success Indicators:**

* All six phases execute sequentially with the documented hard gates.
* Phase 1 checks templates before project materials and WorkIQ evidence.
* Assessment conclusions originate in `rai-plan.md`.
* Missing templates cannot block the workflow; when templates are supplied,
  `assessment-content.md` is required before they are populated.
* Placeholder counts do not limit repeatable rows, widgets, or sticky notes.
  Existing Mural widgets and anchors define layout only; the planner creates
  one widget for every stable-ID content row.
* Without a Mural template, A1 derives from `## System Definition`, A2 from
  `### AI Component Inventory`, and A3 from `## Stakeholder Impact`.
* Recovery preserves consent history and never infers hard-gate confirmation.
* Final documents contain assessment prose rather than drafting or tool
  narration.

For detailed contribution requirements, see:

* Common Standards: [docs/contributing/ai-artifacts-common.md](../docs/contributing/ai-artifacts-common.md) - Shared standards for XML blocks, markdown quality, RFC 2119, validation, and testing
* Agents: [docs/contributing/custom-agents.md](../docs/contributing/custom-agents.md) - Agent configurations with tools and behavior patterns
* Prompts: [docs/contributing/prompts.md](../docs/contributing/prompts.md) - Workflow-specific guidance with template variables
* Instructions: [docs/contributing/instructions.md](../docs/contributing/instructions.md) - Technology-specific standards with glob patterns
* Skills: [docs/contributing/skills.md](../docs/contributing/skills.md) - Task execution utilities with cross-platform scripts

## Testing

All validation ran under Node.js 24:

* [x] `npm run test:ps -- -TestPath scripts/tests/linting/Test-RaiStateSchema.Tests.ps1`
  passed 7 of 7 tests.
* [x] `npm run docs:generate:check`
* [x] `npm run plugin:validate`
* [x] `npm run validate:docs`
* [x] `npm run spell-check`
* [x] `npm run lint:md-links`
* [x] `npm run lint:tables`
* [x] `npm run lint:frontmatter`
* [x] `npm run lint:ai-artifacts`
* [x] `npm run validate:skills`
* [x] `npm run validate:local`

Diff review confirmed that Entry Modes and `suggestedDepthTier` behavior remain
unchanged from the base branch and that placeholder/cardinality behavior is
contained in one commit. The HVE Builder behavior simulation passed with no
in-scope findings. Native Mural, WorkIQ, backlog, signing, and file-write
behavior was not exercised.

### Required Checks

* [x] Documentation is updated
* [x] Files follow existing naming conventions
* [ ] Changes are backwards compatible
* [x] Tests added for new functionality

The six-phase workflow and Entry Modes remain unchanged. The state schema now
requires persisted template-preflight records, so state created before this
change requires migration.

### AI Artifact Contributions

* [x] Used `hve-builder` review mode to review contribution
* [x] Addressed all actionable findings from the `hve-builder` review
* [x] Verified contribution follows common standards and type-specific requirements

### Required Local Checks

The following local-safe validation commands must pass before merging:

* [x] Local validation aggregate: `npm run validate:local`
* [x] Documentation validation (if docs changed): `npm run validate:docs`
* [x] Spell checking: `npm run spell-check`
* [x] Link validation: `npm run lint:md-links`

## Security Considerations

* [x] This PR does not contain any sensitive or NDA information
* [ ] Any new dependencies have been reviewed for security issues (N/A — no dependency changes)
* [ ] Security-related scripts follow the principle of least privilege (N/A — no security script changes)

## GHCP Artifact Maturity

> [!WARNING]
> This PR includes an **experimental** GHCP artifact that may have breaking changes.
>
> * `.github/agents/rai-planning/rai-planner.agent.md`
> * `.github/instructions/rai-planning/rai-identity.instructions.md`

| File                                                             | Type        | Maturity        | Notes            |
|------------------------------------------------------------------|-------------|-----------------|------------------|
| `.github/agents/rai-planning/rai-planner.agent.md`               | Agent       | ⚠️ experimental | Pre-release only |
| `.github/instructions/rai-planning/rai-identity.instructions.md` | Instruction | ⚠️ experimental | Pre-release only |

### GHCP Maturity Acknowledgment

* [x] I acknowledge this PR includes non-stable GHCP artifacts
* [x] Non-stable artifacts are intentional for this change

## Additional Notes

The repository pull request template was preserved. Required AI artifact and
maturity acknowledgments are complete. Nothing has been pushed and no pull
request has been created.

### CODE REVIEW

**Comment by okada-kai** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: .github/instructions/rai-planning/rai-identity.instructions.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Six sequential phases structure the RAI assessment. Each phase declares entry cr
...
...content.md` is mandatory once a template is supplied; failure to create or recover it blocks template population and must be surfaced. `rai-plan.md` remains the authoritative assessment record. Scan `.copilot-tracking/rai-plans/references/` for existing reference content and `.copilot-tracking/rai-plans/{project-slug}/state.json` for existing `referencesProcessed` entries. If existing references are found, present them for confirmation. Otherwise, conduct reference content discovery: ask about evaluation standards and code-of-conduct documents per the User-Supplied Reference Content Protocol and Code-of-Conduct Discovery sections. Confirm the pre-resolved output preferences (outputDetailLevel, targetSystem, audienceProfile, includeOptionalArtifacts). Then proceed with the AI system scoping interview: discover AI system purpose, technology stack, model types, deployment model, stakeholder roles, data inputs, outputs, representativeness, and demographic coverage, intended use contexts, out-of-scope and prohibited use contexts, and autonomous decision boundaries. Classify AI components (model type, training approach, inference pipeline). Establish assessment boundaries and exclusions.
```
Comment (WilliamBerryiii) [inline .github/instructions/rai-planning/rai-identity.instructions.md]: **Medium - instruction density:** Could this be reduced to a short ordered sequence and made the single canonical owner? This line combines the full Phase 1 decision tree, while the same preflight behavior is repeated in the agent. Keeping the protocol here and letting the agent point to it would reduce prompt size and avoid drift.

File: .github/agents/rai-planning/rai-planner.agent.md [hunk 2]
```diff
@@ -41,13 +41,58 @@ When the artifact target matches the telemetry overlay's `applyTo` glob, the ove
...
+template population.
+
 ## Six-Phase Architecture
 
 RAI assessment follows six sequential phases. Each phase collects input through focused questions, prepares artifacts for review, and gates advancement on explicit user confirmation. Phases map to NIST AI RMF functions.
 
 ### Phase 1: AI System Scoping (NIST Govern + Map)
 
-Explore the AI system's purpose, technology stack, deployment model, stakeholder roles, data inputs and outputs, and intended use context. Identify the system's AI components and suggest assessment boundaries. Populate `state.json` with initial project metadata including project slug, entry mode, and AI element inventory. Ask whether the user has specific evaluation standards, risk indicator categories, or output format requirements to incorporate per the User-Supplied Reference Content Protocol in the identity instruction file.
+Begin by resolving and persisting the project slug and output requirements.
```
Comment (WilliamBerryiii) [inline .github/agents/rai-planning/rai-planner.agent.md]: **High - duplicate authority:** This starts a second copy of the Phase 1 preflight protocol already added to the identity instruction. Please replace this detailed block with a concise pointer to the canonical identity and skill guidance so each rule is stated once.

File: scripts/linting/schemas/rai-state.schema.json [hunk 3]
```diff
@@ -51,6 +52,62 @@
...
+            "type": "object",
+            "required": ["kind", "reference"],
+            "properties": {
+              "kind": {
+                "type": "string",
+                "enum": ["document", "mural"]
+              },
+              "reference": {
+                "type": "string",
+                "minLength": 1
```
Comment (WilliamBerryiii) [inline scripts/linting/schemas/rai-state.schema.json]: **High - unsatisfiable persistence contract:** With `additionalProperties: false`, a template record can contain only `kind` and `reference`, but the planner now requires persisting template structure and output requirements. Please represent those required fields here or remove the conflicting prose requirement.

File: scripts/linting/schemas/rai-state.schema.json [hunk 4]
```diff
@@ -51,6 +52,62 @@
...
+        "templates": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": ["kind", "reference"],
+            "properties": {
+              "kind": {
+                "type": "string",
+                "enum": ["document", "mural"]
+              },
```
Comment (WilliamBerryiii) [inline scripts/linting/schemas/rai-state.schema.json:91]: **High - recovery input validation:** This value is dereferenced during recovery but accepts any non-empty string. Please use kind-specific representations: a canonical workspace-contained path for documents and an opaque authenticated ID for Mural. Reject traversal, absolute or UNC paths, URI schemes, and credential-bearing query data before persistence and revalidate on resume.

File: .github/agents/rai-planning/rai-planner.agent.md [hunk 5]
```diff
@@ -70,7 +115,14 @@ Verb sequence:
 3. `mural area list` to resolve A1, A2, A3 by title substring.
 4. `mural tag create` to re-assert the reserved tag manifest (`authored-by-ai`, `rai-phase2`).
 5. `mural area probe` before any parented `mural widget create-bulk` call.
-6. `mural widget create-bulk` per area, decomposing source rows: A1 from the numbered subsections within `## System Definition` in `rai-plan.md`; A2 from the AI component table rows in the `### AI Component Inventory` subsection under `## System Definition`; A3 from bullets in `## Stakeholder Impact`.
+6. `mural widget create-bulk` per area. For a supplied Mural template, use the
+   mandatory `assessment-content.md` stable-ID rows and create one widget for
```
Comment (WilliamBerryiii) [inline .github/agents/rai-planning/rai-planner.agent.md]: **High - incomplete Mural write contract:** Before requiring one widget per row, define each row target area, bound the total write count, and require a sanitized payload preview plus explicit confirmation. Otherwise rows cannot be grouped deterministically and imported evidence can be exported or expanded without a caller-owned safety bound.

File: .github/agents/rai-planning/rai-planner.agent.md [hunk 6]
```diff
@@ -289,7 +354,12 @@ Five-step resume protocol when returning to an existing RAI assessment:
 1. Read `state.json` from the project slug directory.
 2. If `disclaimerShownAt` is `null`, display the Startup Announcement verbatim and set `disclaimerShownAt` to the current ISO 8601 timestamp.
 3. Display current phase progress and checklist status.
-4. Summarize what was completed and what remains.
+4. Read persisted preflight state. When `templates` is non-empty, verify the
+   required `assessmentContentFile`; if it is missing or unusable, pause phase
+   work and recreate it from every recorded template and the authoritative
+   `rai-plan.md`, preserving previously issued stable IDs. Stop and ask the
```
Comment (WilliamBerryiii) [inline .github/agents/rai-planning/rai-planner.agent.md]: **High - stable-ID recovery:** This cannot preserve prior IDs when `assessment-content.md` is the only ID-bearing artifact and is missing. Please persist a recoverable `sourceKey` to `stableId` map outside that file, or stop and obtain confirmation before issuing replacement IDs.

File: .github/agents/rai-planning/rai-planner.agent.md [hunk 7]
```diff
@@ -94,7 +146,7 @@ Facilitate AI-specific threat analysis per component. Catalog potential threats
 
 ### Phase 5: RAI Impact Assessment (NIST Manage)
 
-Explore control surface coverage for each identified threat. Document evidence of existing mitigations and highlight potential gaps. Explore appropriate reliance by examining trust calibration mechanisms, human-in-the-loop design for high-stakes decisions, and patterns of over-reliance or under-reliance. Explore tradeoffs between competing trustworthiness characteristics (for example, transparency versus privacy). Prepare the control surface catalog and evidence register.
+Explore control surface coverage for each identified threat. Document evidence of existing mitigations and highlight potential gaps. Explore appropriate reliance by examining trust calibration mechanisms, human-in-the-loop design for high-stakes decisions, and patterns of over-reliance or under-reliance. Explore tradeoffs between competing trustworthiness characteristics (for example, transparency versus privacy). Prepare the control surface catalog and evidence register. When populating an impact assessment template, add rows or cells when the complete content set exceeds the existing structure.
```
Comment (WilliamBerryiii) [inline .github/agents/rai-planning/rai-planner.agent.md]: **Medium - document accessibility:** When expanding a table, require semantic row and column headers and a logical reading order. "Add rows or cells" alone can produce a visually structured table with no programmatic header relationships for screen-reader users.

File: scripts/tests/fixtures/rai-state/phase-6-complete.json [hunk 8]
```diff
@@ -3,6 +3,19 @@
...
+      {
+        "kind": "document",
+        "reference": "templates/rai-impact-assessment.md"
+      },
+      {
+        "kind": "mural",
+        "reference": "rai-assessment-board-template"
+      }
+    ],
+    "assessmentContentFile": ".copilot-tracking/rai-plans/example-project/assessment-content.md"
```
Comment (WilliamBerryiii) [inline scripts/tests/fixtures/rai-state/phase-6-complete.json:24]: **Medium - boundary coverage:** This exercises only the valid populated-template branch. Please add negative schema tests for a non-empty template list with a null or malformed content path, an empty list with a non-null path, malformed template records, and invalid `securityPlanRef` combinations.

**Review by WilliamBerryiii — CHANGES_REQUESTED** [review]
Thank you for clarifying the template-first RAI workflow. Changes are requested before merge. The Phase 1 protocol is duplicated across the agent and identity instruction and should be consolidated into one concise canonical sequence. The review also found blocking gaps in persisted template metadata, Mural area binding, stable-ID recovery, template-reference validation, and sensitive-data handling before Mural writes. Please address those contracts, add boundary and recovery tests, and align the entry-mode descriptions.

File: .github/instructions/rai-planning/rai-identity.instructions.md (hunk 1 above)
Comment (okada-kai) [reply .github/instructions/rai-planning/rai-identity.instructions.md]: Thank you for the review.
I fixed it here https://github.com/microsoft/hve-core/pull/2764/commits/464658a3aa427226b7565028c5e32dc9f81e3536

File: .github/agents/rai-planning/rai-planner.agent.md (hunk 2 above)
Comment (okada-kai) [reply .github/agents/rai-planning/rai-planner.agent.md]: [464658a](https://github.com/microsoft/hve-core/pull/2764/commits/464658a3aa427226b7565028c5e32dc9f81e3536)

File: scripts/linting/schemas/rai-state.schema.json (hunk 3 above)
Comment (okada-kai) [reply scripts/linting/schemas/rai-state.schema.json]: [66ed814](https://github.com/microsoft/hve-core/pull/2764/commits/66ed814539c3e29ad189b3418bbce5d00cbc5401)

File: scripts/linting/schemas/rai-state.schema.json (hunk 4 above)
Comment (okada-kai) [reply scripts/linting/schemas/rai-state.schema.json:91]: [3073c2d](https://github.com/microsoft/hve-core/pull/2764/commits/3073c2d1e6606d20183c40c01ec77ad39749ebb2)

File: .github/agents/rai-planning/rai-planner.agent.md (hunk 5 above)
Comment (okada-kai) [reply .github/agents/rai-planning/rai-planner.agent.md]: [72868c0](https://github.com/microsoft/hve-core/pull/2764/commits/72868c0fcae25021bf024f06416adc5cbe5f96bf)

File: .github/agents/rai-planning/rai-planner.agent.md (hunk 6 above)
Comment (okada-kai) [reply .github/agents/rai-planning/rai-planner.agent.md]: [66ed814](https://github.com/microsoft/hve-core/pull/2764/commits/66ed814539c3e29ad189b3418bbce5d00cbc5401)

File: .github/agents/rai-planning/rai-planner.agent.md (hunk 7 above)
Comment (okada-kai) [reply .github/agents/rai-planning/rai-planner.agent.md]: [8a9398e](https://github.com/microsoft/hve-core/pull/2764/commits/8a9398e35d2d4381ff17b6d9308a5328fadad894)

File: scripts/tests/fixtures/rai-state/phase-6-complete.json (hunk 8 above)
Comment (okada-kai) [reply scripts/tests/fixtures/rai-state/phase-6-complete.json:24]: [9e18dc1](https://github.com/microsoft/hve-core/pull/2764/commits/9e18dc1592bcc0757193d85bbc1e7dc6f2e227b6)

**Review by jkim323 — APPROVED** [review]

**Review by jkim323 — APPROVED** [review]

**Review by WilliamBerryiii — APPROVED** [review]
